src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+                const auto& nuc {*data::nuclides[j_nuclide]};
+                if (nuc.fissionable_) {
+                  const auto& rxn {*nuc.fission_rx_[0]};
+                  // We need to be careful not to overshoot the number of
+                  // delayed groups since this could cause the range of the
+                  // rxn.products_ array to be exceeded. Hence, we use the size
+                  // of this array and not the MAX_DELAYED_GROUPS constant for
+                  // this loop.
+                  for (auto d = 0; d < rxn.products_.size() - 2; ++d) {
+                    auto yield
+                      = nuc.nu(E, ReactionProduct::EmissionMode::delayed, d+1);
+                    auto rate = rxn.products_[d+1].decay_rate_;
+                    score += p->neutron_xs_[j_nuclide].fission
+                      * yield * atom_density * flux * rate;
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      break;
+
+
+    case SCORE_KAPPA_FISSION:
+      if (p->macro_xs_.absorption == 0.) continue;
+      score = 0.;
+      // Kappa-fission values are determined from the Q-value listed for the
+      // fission cross section.
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        if (settings::survival_biasing) {
+          // No fission events occur if survival biasing is on -- need to
+          // calculate fraction of absorptions that would have resulted in
+          // fission scaled by the Q-value
+          const auto& nuc {*data::nuclides[p->event_nuclide_]};
+          if (p->neutron_xs_[p->event_nuclide_].absorption > 0
+            && nuc.fissionable_) {
+            const auto& rxn {*nuc.fission_rx_[0]};
+            score = p->wgt_absorb_ * rxn.q_value_
+              * p->neutron_xs_[p->event_nuclide_].fission
+              / p->neutron_xs_[p->event_nuclide_].absorption * flux;
+          }
+        } else {
+          // Skip any non-absorption events
+          if (p->event_ == EVENT_SCATTER) continue;
+          // All fission events will contribute, so again we can use particle's
+          // weight entering the collision as the estimate for the fission
+          // reaction rate
+          const auto& nuc {*data::nuclides[p->event_nuclide_]};
+          if (p->neutron_xs_[p->event_nuclide_].absorption > 0
+            && nuc.fissionable_) {
+            const auto& rxn {*nuc.fission_rx_[0]};
+            score = p->wgt_last_ * rxn.q_value_
+              * p->neutron_xs_[p->event_nuclide_].fission
+              / p->neutron_xs_[p->event_nuclide_].absorption * flux;
+          }
+        }
+      } else {
+        if (i_nuclide >= 0) {
+          const auto& nuc {*data::nuclides[i_nuclide]};
+          if (nuc.fissionable_) {
+            const auto& rxn {*nuc.fission_rx_[0]};
+            score = rxn.q_value_ * p->neutron_xs_[i_nuclide].fission
+              * atom_density * flux;
+          }
+        } else {
```

Comment:
Optional: You could simplify this code slightly by collapsing the `else` and `if` blocks.  E.g.:

```
} else if (p->material_ != MATERIAL_VOID) {
  ...
}
```